sdk/cosmos/azure_data_cosmos/src/query/executor.rs

```diff
@@ -76,7 +76,7 @@ impl<T: DeserializeOwned + 'static> QueryExecutor<T> {
                 let query_plan = get_query_plan(
                     &self.http_pipeline,
                     &self.items_link,
-                    self.context.into_borrowed(),
+                    Context::with_context(&self.context),
```

Comment:
Now that I see this, I think I am in favor of adding `into_borrowed()` (just have one call the other, perhaps) but I wouldn't hold up this PR for it. It was definitely cleaner and more intuitive the other way.

Reply:
I believe `with_context` is used in a lot of generated code, so I wanted to avoid changing it in this PR. But I agree `to_borrowed` feels cleaner

Reply:
https://github.com/Azure/azure-sdk-for-rust/issues/2591